base/libgit2/callbacks.jl

```diff
@@ -28,16 +28,8 @@ function authenticate_ssh(creds::SSHCredentials, libgit2credptr::Ptr{Ptr{Void}},
         username_ptr, schema, host)
     isusedcreds = checkused!(creds)
 
-    errcls, errmsg = Error.last_error()
-    if errcls != Error.None
-        # Check if we used ssh-agent
-        if creds.usesshagent == "U"
-            creds.usesshagent = "E" # reported ssh-agent error, disables ssh agent use for the future
```

Comment:
Removing of this did cause the SSH agent only to be used the first time. I've reintroduced this code and updated the helper code to disable using the SSH agent.

The infinite loop scenario I mentioned only appears when working with `credential_loop` which is only used in the tests. There are checks in place within `credential_loop` to abend if this scenario occurs.

Unfortunately, at this time I cannot create tests that utilize the SSH agent so I cannot write tests for the behaviour I broke. However, I did introduce a comment to increase awareness.